Mathlib/AlgebraicTopology/SimplexCategory/GeneratorsRelations/EpiMono.lean

```diff
@@ -98,4 +96,182 @@ lemma eq_or_len_le_of_P_δ {x y : SimplexCategoryGenRel} {f : x ⟶ y} (h_δ : P
 
 end EpiMono
 
+section ExistenceOfFactorizations
+
+/-- An auxiliary lemma to show that one can always use the simplicial identities to simplify a term
+in the form `δ ≫ σ` into either an identity, or a term of the form `σ ≫ δ`. This is the crucial
+special case to induct on to get an epi-mono factorization for all morphisms. -/
+private lemma switch_δ_σ {n : ℕ} (i : Fin (n + 1 + 1)) (i' : Fin (n + 1 + 2)) :
```

Comment:
Adding on the right with "litteral" numbers is "defeq-associative", and Lean is able to match `n + 3` as `m + 1` or `m' + 2` if needed, etc: 
```suggestion
private lemma switch_δ_σ {n : ℕ} (i : Fin (n + 2)) (i' : Fin (n + 3)) :
```